yt/yt/ytlib/api/native/client_operation_info_impl.cpp

```diff
@@ -1269,12 +1324,28 @@ TListOperationsResult TClient::DoListOperations(const TListOperationsOptions& ol
         options.SubstrFilter = to_lower(*options.SubstrFilter);
     }
 
+    auto client = GetOperationsArchiveClient();
+
+    if (Connection_->GetConfig()->StrictSchedulerCommandsAccessValidation) {
+        client = this;
```

Comment:
What does this `if` and assignment mean?

Reply:
It is a client used to create `ObjectServiceReadProxy` to fetch `SubjectClosure (//sys/users/[user]/@member_of_closure)`

By default `OperationsArchiveClient` is used, so I keep it. With `StrictSchedulerCommandsAccessValidation`, `this` is used and it checks that user has `read` access to `//sys/users/[user_for_check]`

Reply:
I would suggest using a ternary operator and adding an NB-comment to clarify this.

Reply:
I would suggest using a ternary operator and adding an NB-comment to clarify this.